crud/drone.py

```diff
@@ -0,0 +1,25 @@
+import sqlalchemy as sa
+from sqlalchemy.orm import sessionmaker
+
+from config import db_config
+from models.drone import DroneStatus
+
+
+db = f"""postgresql://\
+{db_config.POSTGRES_USER}:{db_config.POSTGRES_PASSWORD}\
+@{db_config.POSTGRES_HOST}:{db_config.POSTGRES_OUT_PORT}/{db_config.POSTGRES_DB}"""
+engine = sa.create_engine(db)
+Session = sessionmaker(bind=engine)
+connection = engine.connect()
+session = Session()
```

Comment:
We need to reorganize your code, using Session instead of a lot of connections.
1. Create a crud/session.py file:
```
from*** import settings
from simba_nft.discoverability import nft_tracing
from sqlalchemy.ext.asyncio import AsyncSession, create_async_engine
from sqlalchemy.orm import sessionmaker


engine = create_async_engine(
    settings.SQLALCHEMY_DATABASE_URI, # db in your case
    pool_pre_ping=True,
    pool_size=5,
    max_overflow=10,
    echo=False,
)

SessionLocal = sessionmaker(
    autocommit=False,
    autoflush=False,
    bind=engine,
    class_=AsyncSession,
    expire_on_commit=False,
)
```
This is how you describe the connection.

2. Then you re-format all yours crud/*.py files to include method(s) to a class, for example:
instead of 
```
...
async def read_data_for_new_task() -> Task:
     ...
    return next_task
async def create_task_in_db(add_latitude, add_longitude, add_priority) -> Task:
      ....
      return ...
```
you need to use
```
...

class Station():
async def read_data_for_new_task() -> Task:
     ...
    return next_task
async def create_task_in_db(add_latitude, add_longitude, add_priority) -> Task:
      ....
      return ...
station = Station()
```
BTW! Change filenames to crud_*.py - will be easy to recognize and you can avoid problems with import!

3. Then you need to change crud/__init__py:
import wrapt
from .crud_station import station
async def get_db() -> AsyncGenerator[AsyncSession, None]:
    ### https://docs.sqlalchemy.org/en/14/orm/session_transaction.html#managing-transactions
    async with SessionLocal() as session:
        yield session
        try:
            await session.commit()
        except Exception:
            await session.rollback()

@wrapt.decorator
async def rollback_db_on_error(
    wrapped: Any, instance: Any, args: list, kwargs: dict
) -> Any:
    session: AsyncSession = kwargs["db"]
    try:
        return await wrapped(*args, **kwargs)
    except Exception:
        await session.rollback()
        raise
```

4. And the last one - you need to set the first parameter of your endpoint:
```
@router.post('/add_coord', response_model=TaskBase)
async def add_task_to_db(
   db: AsyncSession = Depends(get_db),
    task: Task
) -> TaskBase:
```
and then you need to send "db" parameter to ANY function that will use the db